src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/dynamiclists/DynamicLetters.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.hmcts.ethos.replacement.docmosis.helpers.dynamiclists;
+
+import uk.gov.hmcts.ecm.common.model.bulk.types.DynamicFixedListType;
+import uk.gov.hmcts.ecm.common.model.bulk.types.DynamicValueType;
+import uk.gov.hmcts.ecm.common.model.ccd.CaseData;
+import uk.gov.hmcts.ecm.common.model.ccd.SubmitEvent;
+import uk.gov.hmcts.ecm.common.model.ccd.types.CorrespondenceScotType;
+import uk.gov.hmcts.ecm.common.model.ccd.types.CorrespondenceType;
+import uk.gov.hmcts.ecm.common.model.multiples.MultipleData;
+import uk.gov.hmcts.ethos.replacement.docmosis.helpers.DynamicListHelper;
+
+import java.util.List;
+
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.SCOTLAND_BULK_CASE_TYPE_ID;
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.SCOTLAND_CASE_TYPE_ID;
+
+public class DynamicLetters {
+    private DynamicLetters() {
+    }
+
+    public static void dynamicLetters(CaseData caseData, String caseTypeId) {
+        List<DynamicValueType> listItems = DynamicListHelper.createDynamicHearingList(caseData);
+        if (!listItems.isEmpty()) {
```

Comment:
if listItems is null, listItems.isEmpty will throw null pointer exception. How about using CollectionUtils.isEmpty?

Reply:
Refactored

Reply:
Even better do not return a null collection!

Reply:
The null checks on collections we have are really due to the CCD data imposed on us - where we are writing methods that return data they should never return null

Reply:
So I feel this code is correct

Reply:
It would have returned a blank list rather than null but have changed it anyway